From 0e5edf7f0cdcbe5f34d98aeab5ee4c6a2259754e Mon Sep 17 00:00:00 2001
From: Akshay Pradip Bharad <apbharad@habana.ai>
Date: Wed, 21 May 2025 11:23:32 +0000
Subject: [PATCH] [SW-0] gaudi3: interrupt: remove redundant argument

Function argument 'action' is not used in individual component
handlers.
This patch removes all the instances of this redundant argument.

Change-Id: Ie742af3be9d485070269d6b28c7e9b806c5f05c6
Signed-off-by: Akshay Pradip Bharad <apbharad@habana.ai>
---

diff --git a/src/gaudi3/interrupt.c b/src/gaudi3/interrupt.c
index ae858cc..4cd0e16 100644
--- a/src/gaudi3/interrupt.c
+++ b/src/gaudi3/interrupt.c
@@ -609,8 +609,7 @@
 #define MME_CTRL_SEI0_QM_INTR_MASK	0x12
 #define MME_CTRL_SEI1_QM_INTR_MASK	0x40
 static void gaudi3_process_mme_spi_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t offset, base = 0, val, idx;
 	uint32_t spmu_offset, bmon_offset;
@@ -742,8 +741,7 @@
 }
 
 static void gaudi3_process_mme_sei_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t offset, base = 0, val, idx, mask;
 	uint32_t qm_base, arc_base;
@@ -1109,8 +1107,7 @@
 
 static void gaudi3_process_tpc_spi_interrupt(struct hl_event_config *config,
 					     struct gaudi3_async_events_ids_map *intr_map,
-					     struct gaudi3_async_events_ids_map *binned_intr_map,
-					     enum hl_intr_action action)
+					     struct gaudi3_async_events_ids_map *binned_intr_map)
 {
 	struct hl_eq_tpc_spi_data tpc_spi_data;
 	uint32_t offset, base = 0;
@@ -1165,8 +1162,7 @@
 
 static void gaudi3_process_tpc_sei_interrupt(struct hl_event_config *config,
 					     struct gaudi3_async_events_ids_map *intr_map,
-					     struct gaudi3_async_events_ids_map *binned_intr_map,
-					     enum hl_intr_action action)
+					     struct gaudi3_async_events_ids_map *binned_intr_map)
 {
 	switch (binned_intr_map->index) {
 	case SEI_INTR_TPC0_AXI_SLV_DEC_ERR:
@@ -1285,8 +1281,7 @@
 }
 
 static void gaudi3_process_rot_spi_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t base = 0, cs_dbg_base, cause, intr_clr;
 	uint32_t spmu_offset, bmon_offset;
@@ -1349,8 +1344,7 @@
 
 static void gaudi3_process_rot_sei_interrupt(struct hl_event_config *config,
 					struct gaudi3_async_events_ids_map *intr_map,
-					struct gaudi3_async_events_ids_map *binned_intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *binned_intr_map)
 {
 	struct hl_eq_rot_sei_data data = { 0 };
 	uint32_t qm_base, arc_base;
@@ -1400,8 +1394,7 @@
 }
 
 static void gaudi3_process_cs_spi_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t cs_base = 0, cs_dbg_base = 0;
 	uint32_t cs_cause;
@@ -1531,8 +1524,7 @@
 }
 
 static void gaudi3_process_cs_sei_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_cs_sei_data data;
 	uint32_t base;
@@ -1620,8 +1612,7 @@
 }
 
 static void gaudi3_process_stlb_spi_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t base = 0, syndrom_l, syndrom_h, mask = 0;
 	struct hl_eq_stlb_spi_data data = { 0 };
@@ -1664,8 +1655,7 @@
 }
 
 static void gaudi3_process_stlb_sei_interrupt(struct hl_event_config *config,
-					      struct gaudi3_async_events_ids_map *intr_map,
-					      enum hl_intr_action action)
+					      struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_stlb_sei_data data = { 0 };
 	uint32_t base, syndrom_l, syndrom_h;
@@ -1705,8 +1695,7 @@
 }
 
 static void gaudi3_process_edma_sob_spi_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t base, spmu_offset, bmon_offset;
 	struct hl_eq_generic_spi_data data;
@@ -1765,8 +1754,7 @@
 }
 
 static void gaudi3_process_sob_sei_interrupt(struct hl_event_config *config,
-					     struct gaudi3_async_events_ids_map *intr_map,
-					     enum hl_intr_action action)
+					     struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_sob_sei_data data = {0};
 	uint32_t base, cq_intr_val;
@@ -1815,8 +1803,7 @@
 #define EDMA_BASE_DIFF		(mmHD1_SEDMA1_QM_DCCM_BASE - mmHD1_SEDMA0_QM_DCCM_BASE)
 
 static void gaudi3_process_edma_sei_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t qm_base, arc_base, ch0_base, offset, idx;
 	struct hl_eq_edma_sei_data data;
@@ -1980,8 +1967,7 @@
 (mmD1_HBM0_MC01_CS_DBG_ROM_TABLE_BASE - mmD0_HBM0_MC01_CS_DBG_ROM_TABLE_BASE)
 
 static void gaudi3_process_mc_spi_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t base, cs_dbg_base, spmu_offset, intr_offset;
 	struct hl_eq_generic_spi_data data;
@@ -2541,8 +2527,7 @@
 }
 
 static void gaudi3_process_mc_sei_interrupt(struct hl_event_config *config,
-					    struct gaudi3_async_events_ids_map *intr_map,
-					    enum hl_intr_action action)
+					    struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint8_t ecc_derr_err = 0, ecc_serr_err = 0, rd_par_err = 0, ca_par_err = 0, wr_par_err = 0;
 	uint32_t hbm_dev, mc_cmn_idx, hbm_offset, mc_cmn_offset, sei0_sts, sei1_sts;
@@ -2646,8 +2631,7 @@
 }
 
 static void gaudi3_process_dec_spi_interrupt(struct hl_event_config *config,
-					struct gaudi3_async_events_ids_map *intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t base, cmd_base, spmu_offset, bmon_offset, mask;
 	struct hl_eq_generic_spi_data dec_data;
@@ -2709,8 +2693,7 @@
 
 static void gaudi3_process_dec_sei_interrupt(struct hl_event_config *config,
 					struct gaudi3_async_events_ids_map *intr_map,
-					struct gaudi3_async_events_ids_map *binned_intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *binned_intr_map)
 {
 	struct hl_eq_razwi_with_intr_cause_data data = { 0 };
 	uint32_t base;
@@ -2754,8 +2737,7 @@
 #define FARM_ARC1_SEI_INTR_STS_VAL_M		BIT(1)
 
 static void gaudi3_process_arcfarm_sei_interrupt(struct hl_event_config *config,
-						 struct gaudi3_async_events_ids_map *intr_map,
-						 enum hl_intr_action action)
+						 struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t base, arc0_base = 0, arc1_base = 0, arc_sts;
 	struct hl_eq_arcfarm_sei_data data;
@@ -2819,8 +2801,7 @@
 }
 
 static void gaudi3_process_edup_sei_interrupt(struct hl_event_config *config,
-					      struct gaudi3_async_events_ids_map *intr_map,
-					      enum hl_intr_action action)
+					      struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_razwi_with_intr_cause_data data = { 0 };
 	uint32_t base;
@@ -2854,8 +2835,7 @@
 }
 
 static void gaudi3_process_psoc_sei_interrupt(struct hl_event_config *config,
-					      struct gaudi3_async_events_ids_map *intr_map,
-					      enum hl_intr_action action)
+					      struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_psoc_sei_data data = {0};
 	uint32_t base = mmD0_PSOC_GLOBAL_CONF2_BASE + (DIE_OFFSET * intr_map->die_id);
@@ -2877,8 +2857,7 @@
 }
 
 static void gaudi3_process_parc_sei_interrupt(struct hl_event_config *config,
-					      struct gaudi3_async_events_ids_map *intr_map,
-					      enum hl_intr_action action)
+					      struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_parc_sei_data data = {0};
 	uint32_t base = mmD0_PARC_GLOBAL_CONF_BASE + (DIE_OFFSET * intr_map->die_id);
@@ -2900,8 +2879,7 @@
 }
 
 static void gaudi3_process_nic_spmu_bmon(struct hl_event_config *config,
-						struct gaudi3_async_events_ids_map *intr_map,
-						enum hl_intr_action action)
+						struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t spmu_offset, bmon_offset, nic_offset, die_offset;
 	struct hl_eq_nic_spi_data data = {0};
@@ -2920,8 +2898,7 @@
 }
 
 static void gaudi3_process_nch_spi_interrupt(struct hl_event_config *config,
-					     struct gaudi3_async_events_ids_map *intr_map,
-					     enum hl_intr_action action)
+					     struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_intr_cause data = {0};
 	uint32_t base = mmD0_NCH_BASE + (DIE_OFFSET * intr_map->die_id);
@@ -2937,8 +2914,7 @@
 }
 
 static void gaudi3_process_nic_sw_error(struct hl_event_config *config,
-					     struct gaudi3_async_events_ids_map *intr_map,
-					     enum hl_intr_action action)
+					     struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t qpc_base, rxb_core_base, rxe_base, nic_offset, die_offset;
 	struct hl_eq_nic_spi_data data = {0};
@@ -2994,8 +2970,7 @@
 }
 
 static void gaudi3_process_nic_spi_interrupt(struct hl_event_config *config,
-					     struct gaudi3_async_events_ids_map *intr_map,
-					     enum hl_intr_action action)
+					     struct gaudi3_async_events_ids_map *intr_map)
 {
 
 	switch (intr_map->index) {
@@ -3005,7 +2980,7 @@
 	case SPI_INTR_SHARED_NIC3_CS_DBG:
 	case SPI_INTR_SHARED_NIC4_CS_DBG:
 	case SPI_INTR_SHARED_NIC5_CS_DBG:
-		gaudi3_process_nic_spmu_bmon(config, intr_map, action);
+		gaudi3_process_nic_spmu_bmon(config, intr_map);
 		break;
 	case SPI_INTR_SHARED_NIC0_RXB_QPC_RXE:
 	case SPI_INTR_SHARED_NIC1_RXB_QPC_RXE:
@@ -3013,7 +2988,7 @@
 	case SPI_INTR_SHARED_NIC3_RXB_QPC_RXE:
 	case SPI_INTR_SHARED_NIC4_RXB_QPC_RXE:
 	case SPI_INTR_SHARED_NIC5_RXB_QPC_RXE:
-		gaudi3_process_nic_sw_error(config, intr_map, action);
+		gaudi3_process_nic_sw_error(config, intr_map);
 		break;
 	default:
 		LOG_ERR("Invalid interrupt index %d received for comp_type: %d",
@@ -3023,8 +2998,7 @@
 }
 
 static void gaudi3_process_nic_sei_interrupt(struct hl_event_config *config,
-					     struct gaudi3_async_events_ids_map *intr_map,
-					     enum hl_intr_action action)
+					     struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t qpc_base, tmr_base, rxb_core_base, rxe_base, txs_base, txe_base;
 	struct hl_eq_nic_sei_data data = {0};
@@ -3265,8 +3239,7 @@
 }
 
 static void gaudi3_process_pcie_sei_interrupt(struct hl_event_config *config,
-					      struct gaudi3_async_events_ids_map *intr_map,
-					      enum hl_intr_action action)
+					      struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_pcie_sei_data data = { 0 };
 	uint32_t base, dummy_read;
@@ -3329,8 +3302,7 @@
 }
 
 static void gaudi3_process_pmmu_sei_interrupt(struct hl_event_config *config,
-					      struct gaudi3_async_events_ids_map *intr_map,
-					      enum hl_intr_action action)
+					      struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_razwi_with_intr_cause_data data = { 0 };
 	uint32_t base = 0;
@@ -3377,8 +3349,7 @@
 #define SPDMA_OFFSET	(mmD0_SPDMA1_CH0_A_BASE - mmD0_SPDMA0_CH0_A_BASE)
 
 static void gaudi3_process_pdma_sei_interrupt(struct hl_event_config *config,
-					      struct gaudi3_async_events_ids_map *intr_map,
-					      enum hl_intr_action action)
+					      struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_pdma_sei_data data = { 0 };
 	uint8_t i, j;
@@ -3450,8 +3421,7 @@
 }
 
 static void gaudi3_process_cpu_sei_interrupt(struct hl_event_config *config,
-					     struct gaudi3_async_events_ids_map *intr_map,
-					     enum hl_intr_action action)
+					     struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_cpu_sei_data data = { 0 };
 	uint32_t base;
@@ -3487,8 +3457,7 @@
 }
 
 static void gaudi3_process_nch_sei_interrupt(struct hl_event_config *config,
-					     struct gaudi3_async_events_ids_map *intr_map,
-					     enum hl_intr_action action)
+					     struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_nch_sei_data data = { 0 };
 
@@ -3598,8 +3567,7 @@
 #endif
 
 static void gaudi3_process_d2dmac_sei_interrupt(struct hl_event_config *config,
-						struct gaudi3_async_events_ids_map *intr_map,
-						enum hl_intr_action action)
+						struct gaudi3_async_events_ids_map *intr_map)
 {
 #ifndef TARGET_IS_ONIX
 	int ret;
@@ -3640,8 +3608,7 @@
 }
 
 static void gaudi3_process_d2dphy_sei_interrupt(struct hl_event_config *config,
-						struct gaudi3_async_events_ids_map *intr_map,
-						enum hl_intr_action action)
+						struct gaudi3_async_events_ids_map *intr_map)
 {
 #ifndef TARGET_IS_ONIX
 	int ret;
@@ -3704,8 +3671,7 @@
 };
 
 static void gaudi3_process_pll_sei_interrupt(struct hl_event_config *config,
-						struct gaudi3_async_events_ids_map *intr_map,
-						enum hl_intr_action action)
+						struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_pll_sei_data data = {0};
 	uint32_t base;
@@ -3842,8 +3808,7 @@
 #define BMU_REGION_HIT_UNMASK	0x0
 
 static void gaudi3_process_pcie_spi_interrupt(struct hl_event_config *config,
-					      struct gaudi3_async_events_ids_map *intr_map,
-					      enum hl_intr_action action)
+					      struct gaudi3_async_events_ids_map *intr_map)
 {
 	struct hl_eq_pcie_spi_data data = { 0 };
 	uint32_t base, dbi_access_base, spmu_offset, bmon_offset, die_offset;
@@ -4093,8 +4058,7 @@
 #define SPI_INTR_SHARED_PMMU2_MASK	0x8
 
 static void gaudi3_process_pmmu_spi_interrupt(struct hl_event_config *config,
-					      struct gaudi3_async_events_ids_map *intr_map,
-					      enum hl_intr_action action)
+					      struct gaudi3_async_events_ids_map *intr_map)
 {
 	uint32_t base = mmD0_PMMU_HBW_MMU_BASE + (DIE_OFFSET * intr_map->die_id);
 	uint32_t axi_id1, axi_id2, lsb_va, msb_va, err_type, cause,
@@ -4174,46 +4138,45 @@
 
 static void gaudi3_process_spi_interrupt(struct hl_event_config *config,
 					struct gaudi3_async_events_ids_map *intr_map,
-					struct gaudi3_async_events_ids_map *binned_intr_map,
-					enum hl_intr_action action)
+					struct gaudi3_async_events_ids_map *binned_intr_map)
 {
 	switch (intr_map->comp_type) {
 	case INT_COMP_TYPE_MME:
-		gaudi3_process_mme_spi_interrupt(config, intr_map, action);
+		gaudi3_process_mme_spi_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_TPC:
-		gaudi3_process_tpc_spi_interrupt(config, intr_map, binned_intr_map, action);
+		gaudi3_process_tpc_spi_interrupt(config, intr_map, binned_intr_map);
 		break;
 	case INT_COMP_TYPE_ROT:
-		gaudi3_process_rot_spi_interrupt(config, binned_intr_map, action);
+		gaudi3_process_rot_spi_interrupt(config, binned_intr_map);
 		break;
 	case INT_COMP_TYPE_CS:
-		gaudi3_process_cs_spi_interrupt(config, intr_map, action);
+		gaudi3_process_cs_spi_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_STLB:
-		gaudi3_process_stlb_spi_interrupt(config, intr_map, action);
+		gaudi3_process_stlb_spi_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_EDMA:
 	case INT_COMP_TYPE_SOB:
-		gaudi3_process_edma_sob_spi_interrupt(config, intr_map, action);
+		gaudi3_process_edma_sob_spi_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_MC:
-		gaudi3_process_mc_spi_interrupt(config, intr_map, action);
+		gaudi3_process_mc_spi_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_DEC:
-		gaudi3_process_dec_spi_interrupt(config, binned_intr_map, action);
+		gaudi3_process_dec_spi_interrupt(config, binned_intr_map);
 		break;
 	case INT_COMP_TYPE_NIC:
-		gaudi3_process_nic_spi_interrupt(config, intr_map, action);
+		gaudi3_process_nic_spi_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_PMMU:
-		gaudi3_process_pmmu_spi_interrupt(config, intr_map, action);
+		gaudi3_process_pmmu_spi_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_PCIE:
-		gaudi3_process_pcie_spi_interrupt(config, intr_map, action);
+		gaudi3_process_pcie_spi_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_NCH:
-		gaudi3_process_nch_spi_interrupt(config, intr_map, action);
+		gaudi3_process_nch_spi_interrupt(config, intr_map);
 		break;
 	default:
 		break;
@@ -4222,75 +4185,74 @@
 
 static void gaudi3_process_sei_interrupt(struct hl_event_config *config,
 					 struct gaudi3_async_events_ids_map *intr_map,
-					 struct gaudi3_async_events_ids_map *binned_intr_map,
-					 enum hl_intr_action action)
+					 struct gaudi3_async_events_ids_map *binned_intr_map)
 {
 	switch (intr_map->comp_type) {
 	case INT_COMP_TYPE_MME:
-		gaudi3_process_mme_sei_interrupt(config, intr_map, action);
+		gaudi3_process_mme_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_TPC:
-		gaudi3_process_tpc_sei_interrupt(config, intr_map, binned_intr_map, action);
+		gaudi3_process_tpc_sei_interrupt(config, intr_map, binned_intr_map);
 		break;
 	case INT_COMP_TYPE_ROT:
-		gaudi3_process_rot_sei_interrupt(config, intr_map, binned_intr_map, action);
+		gaudi3_process_rot_sei_interrupt(config, intr_map, binned_intr_map);
 		break;
 	case INT_COMP_TYPE_CS:
-		gaudi3_process_cs_sei_interrupt(config, intr_map, action);
+		gaudi3_process_cs_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_STLB:
-		gaudi3_process_stlb_sei_interrupt(config, intr_map, action);
+		gaudi3_process_stlb_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_EDMA:
-		gaudi3_process_edma_sei_interrupt(config, intr_map, action);
+		gaudi3_process_edma_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_SOB:
-		gaudi3_process_sob_sei_interrupt(config, intr_map, action);
+		gaudi3_process_sob_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_MC:
-		gaudi3_process_mc_sei_interrupt(config, intr_map, action);
+		gaudi3_process_mc_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_DEC:
-		gaudi3_process_dec_sei_interrupt(config, intr_map, binned_intr_map, action);
+		gaudi3_process_dec_sei_interrupt(config, intr_map, binned_intr_map);
 		break;
 	case INT_COMP_TYPE_ARC_FARM:
-		gaudi3_process_arcfarm_sei_interrupt(config, intr_map, action);
+		gaudi3_process_arcfarm_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_EDUP:
-		gaudi3_process_edup_sei_interrupt(config, intr_map, action);
+		gaudi3_process_edup_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_PSOC:
-		gaudi3_process_psoc_sei_interrupt(config, intr_map, action);
+		gaudi3_process_psoc_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_PARC:
-		gaudi3_process_parc_sei_interrupt(config, intr_map, action);
+		gaudi3_process_parc_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_NIC:
-		gaudi3_process_nic_sei_interrupt(config, intr_map, action);
+		gaudi3_process_nic_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_PCIE:
-		gaudi3_process_pcie_sei_interrupt(config, intr_map, action);
+		gaudi3_process_pcie_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_PMMU:
-		gaudi3_process_pmmu_sei_interrupt(config, intr_map, action);
+		gaudi3_process_pmmu_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_PDMA:
-		gaudi3_process_pdma_sei_interrupt(config, intr_map, action);
+		gaudi3_process_pdma_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_CPU:
-		gaudi3_process_cpu_sei_interrupt(config, intr_map, action);
+		gaudi3_process_cpu_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_NCH:
-		gaudi3_process_nch_sei_interrupt(config, intr_map, action);
+		gaudi3_process_nch_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_D2D_MAC:
-		gaudi3_process_d2dmac_sei_interrupt(config, intr_map, action);
+		gaudi3_process_d2dmac_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_D2D_PHY:
-		gaudi3_process_d2dphy_sei_interrupt(config, intr_map, action);
+		gaudi3_process_d2dphy_sei_interrupt(config, intr_map);
 		break;
 	case INT_COMP_TYPE_PLL:
-		gaudi3_process_pll_sei_interrupt(config, intr_map, action);
+		gaudi3_process_pll_sei_interrupt(config, intr_map);
 		break;
 	default:
 		LOG_ERR("Interrupt index %d not supported  for comp_type: %d",
@@ -4599,10 +4561,10 @@
 		gaudi3_process_rei_interrupt(config, &binned_intr_map);
 		break;
 	case INT_GRP_TYPE_SEI:
-		gaudi3_process_sei_interrupt(config, &intr_map, &binned_intr_map, action);
+		gaudi3_process_sei_interrupt(config, &intr_map, &binned_intr_map);
 		break;
 	case INT_GRP_TYPE_SPI:
-		gaudi3_process_spi_interrupt(config, &intr_map, &binned_intr_map, action);
+		gaudi3_process_spi_interrupt(config, &intr_map, &binned_intr_map);
 		break;
 	case INT_GRP_TYPE_ECO:
 		break;
